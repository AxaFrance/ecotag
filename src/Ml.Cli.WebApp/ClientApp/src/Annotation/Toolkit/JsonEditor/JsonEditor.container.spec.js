import React from "react";
import {render, waitFor, fireEvent} from '@testing-library/react';
import {Textarea} from "@axa-fr/react-toolkit-form-input-textarea";
import JsonEditorContainer from "./JsonEditor.container";

const expectedResult = {};

const mockedMonacoEditor = ({id, height, language, value, onChange, options}) => {
    return (
        <Textarea
            id={id}
            name="Textarea"
            value={value}
            onChange={onChange}
        />
    );
};

describe('Check Editor usage', () => {
    test('Change Editor content', async () => {

        let hasBeenClicked = false;

        const {container, asFragment, getAllByText} = render(
            <JsonEditorContainer
                expectedOutput={{
                    id: "Editor-container-test",
                    fileName: "fileName_value",
                    value: "{\"analysis\": \"This is a test\"}"
                }}
                urls={[]}
                onSubmit={() => hasBeenClicked = true}
                MonacoEditor={mockedMonacoEditor}
            />
        );
        await waitFor(() => expect(getAllByText(/Aucun fichier correspondant n'a été trouvé sur votre disque./i)).not.toBeNull());
        const fileInput = container.querySelector("#Editor-container-test");
        expect(fileInput).toHaveTextContent("{ \"analysis\": \"This is a test\" }");
        const saveButton = container.querySelector(".btn.af-btn");
        fireEvent.click(saveButton);
        expect(hasBeenClicked).toEqual(true);
        expect(asFragment()).toMatchSnapshot();
    });
});
