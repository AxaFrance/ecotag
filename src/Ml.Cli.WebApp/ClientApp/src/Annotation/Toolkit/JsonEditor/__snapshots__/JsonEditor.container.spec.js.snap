// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Editor usage Change Editor content 1`] = `
<DocumentFragment>
  <div
    class="images-editor-container"
  >
    <div
      class="images-editor-container__images_column"
    >
      <div
        class="images-error"
      >
        <p>
          No related file has been found on your local hard drive.
          <br />
           There is several possibilities:
          <br />
           - There is no existing image related to this file on your hard drive;
          <br />
           - The specified regex doesn't match any file;
          <br />
           - The images repository of this file doesn't exist.
        </p>
      </div>
      <div
        class="images-list"
      />
    </div>
    <div
      class="editor-container"
    >
      <textarea
        class="af-form__input-textarea"
        id="Editor-container-test"
        name="Textarea"
        placeholder=""
      >
        {
  "analysis": "This is a test"
}
      </textarea>
      <div
        class="editor-container__errors-container"
      />
      <div
        class="editor-container__button"
      >
        <button
          class="btn af-btn"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
